src/diart/websockets.py

```diff
@@ -0,0 +1,298 @@
+        except Exception as e:
+            logger.error(f"Error processing message from client {client_id}: {e}")
+            # Don't close the connection for non-connection related errors
+            # This allows the client to retry sending the message
+
+    def send(self, client_id: Text, message: AnyStr) -> None:
+        """Send a message to a specific client.
+
+        Parameters
+        ----------
+        client_id : Text
+            Target client identifier
+        message : AnyStr
+            Message to send
+        """
+        if not message:
+            return
+
+        client = next((c for c in self.server.clients if c["id"] == client_id), None)
+
+        if client is None:
+            return
+
+        try:
+            self.server.send_message(client, message)
+        except (socket.error, ConnectionError) as e:
+            logger.warning(
+                f"Client {client_id} disconnected while sending message: {e}"
+            )
+            self.close(client_id)
+        except Exception as e:
+            logger.error(f"Failed to send message to client {client_id}: {e}")
+
+    def run(self) -> None:
+        """Start the WebSocket server."""
+        logger.info(f"Starting WebSocket server on {self.uri}")
+        max_retries = 3
+        retry_count = 0
+
+        while retry_count < max_retries:
+            try:
+                self.server.run_forever()
+                break  # If server exits normally, break the retry loop
+            except (socket.error, ConnectionError) as e:
+                logger.warning(f"WebSocket connection error: {e}")
+                retry_count += 1
+                if retry_count < max_retries:
+                    logger.info(
+                        f"Attempting to restart server (attempt {retry_count + 1}/{max_retries})"
+                    )
+                else:
+                    logger.error("Max retry attempts reached. Server shutting down.")
+            except Exception as e:
+                logger.error(f"Fatal server error: {e}")
+                break
+            finally:
+                self.close_all()
+
+    def close(self, client_id: Text) -> None:
+        """Close a specific client's connection and cleanup resources.
+
+        Parameters
+        ----------
+        client_id : Text
+            Client identifier to close
+        """
+        if client_id not in self._clients:
+            return
+
+        try:
+            # Clean up pipeline state using built-in reset method
+            client_state = self._clients[client_id]
+            client_state.inference.pipeline.reset()
+
+            # Close audio source and remove client
+            client_state.audio_source.close()
+            del self._clients[client_id]
+
+            # Try to send a close frame to the client
+            client = next((c for c in self.server.clients if c["id"] == client_id), None)
+
+            if client is None:
+                return
```

Comment:
Also add a log here. Can this happen btw?

Reply:
from docstring of `send` method:
```
Raises
------
Warning
    If client is not found in server.clients. Common cases:
    - Client disconnected but cleanup is not complete
    - Network issues caused client to drop
    - Race condition between disconnect and message send
```